pythonopensubtitles/opensubtitles.py

```diff
@@ -104,9 +109,99 @@ def detect_language(self):
         # array DetectLanguage( $token, array($text, $text, ...) )
         raise NotImplementedError
 
-    def download_subtitles(self):
-        # array DownloadSubtitles( $token, array($IDSubtitleFile, $IDSubtitleFile,...) )
-        raise NotImplementedError
+    def download_subtitles(self, subfile_ids, ext='srt',
+                           filenames=None, directory=None):
+        """
+        Returns a dictionary with max. 20 IDs of and paths to all
+        successfully downloaded subtitle files, otherwise None.
+
+        subfile_ids can be None/empty if a filenames dict is present;
+        otherwise, the file IDs used by OpenSubtitles are used as file
+        names, with .srt as default file extension.
+
+        Keep in mind that subtitle searches can contain multiple
+        results for the same file title (because several different
+        file uploads might reference the same file name).
+        So, if you use a filenames dict in which different file IDs
+        reference the exact same file name (i.e. IDs with the same value),
+        the number of eventually downloaded files will be smaller than the
+        overall number of file IDs submitted (there can only be one file
+        per file name, so any subsequent download attempts for an already
+        existing file will overwrite the previous version of said file).
+
+        :param subfile_ids: list of ids of subtitle files to download
+        :param filenames: optional dictionary with preferred file names;
+                          for keys use subtitle file IDs (as strings),
+                          for values use file names (incl. file extensions)
+        :param ext: one of: srt, sub, txt, ssa, smi, mpi, tmp;
+                    only applicable if only subtitle file IDs are provided
+        :param directory: path to preferred dir to which to write files
+        """
+        successful = {}
+        if filenames:
+            subfile_ids = list(filenames.keys())
+
+        # OpenSubtitles will accept a maximum of 20 IDs for download
+        if len(subfile_ids) > 20:
+            print("Cannot download more than 20 files at once.",
+                  file=sys.stderr)
+            subfile_ids = subfile_ids[:20]
+
+        self.data = self.xmlrpc.DownloadSubtitles(self.token, subfile_ids)
+        encoded_data = self._get_from_data_or_none('data')
+
+        if encoded_data:
```

Comment:
I would negate this if so you can `return` first or you can just raise an exception. This way you would avoid one indentation level.

Reply:
You mean:
```
if not encoded data:
    return

... rest of code here... 
```

I think I had it like this at first, but then switched it around because I wanted to use the same `None` for when a) there is data but no successful downloads or b) when there is no data at all. (Which I think is shorter than to repeat the check for the `successful` variable later on the way I would write it, with if/else, but probably the same if I do the `return ... or None` like you do it.)